pkg/controller/servicebindingrequest/annotations/annotations.go

```diff
@@ -0,0 +1,76 @@
+package annotations
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+// bindingType encodes the medium the binding should deliver the configuration value.
+type bindingType string
+
+const (
+	// BindingTypeVolumeMount indicates the binding should happen through a volume mount.
+	BindingTypeVolumeMount bindingType = "volumemount"
+	// BindingTypeEnvVar indicates the binding should happen through environment variables.
+	BindingTypeEnvVar bindingType = "env"
+)
+
+// supportedBindingTypes contains all currently supported binding types.
+var supportedBindingTypes = map[bindingType]bool{
+	BindingTypeVolumeMount: true,
+	BindingTypeEnvVar:      true,
+}
+
+// dataPath is the path ConfigMap and Secret resources use to scope their data.
+//
+// note: it is currently used to provide a pointer to the "data" string, which is the location
+// ConfigMap and Secret resources keep user data.
+var dataPath = "data"
+
+// Result contains data that has been collected by an annotation handler.
+type Result struct {
+	// Object is the data that has been collected by the Handler; it can be found in the location
+	// indicated by the Path field.
+	Object map[string]interface{}
+	// Type indicates where the Object field should be injected in the application; can be either
+	// "env" or "volumemount".
+	Type bindingType
+	// Path is the nested location the collected data can be found in the Object field.
+	Path string
```

Comment:
I am confused here - are we collecting data in `Object` field, or we are collecting more than what needed, and `Path` gives us hint where are the data we are interested in? Could we just put in `Object` what we refer via `Path` field and drop this field completely.

More general question is what does `Result `model?

Reply:
`Path` is only used in [servicecontext.go#L130-L132](https://github.com/isutton/service-binding-operator/blob/1b517019a07b21453f8f62aef8df3ff769d5fc50/pkg/controller/servicebindingrequest/servicecontext.go#L130-L132) to compose the volume keys list because it's impossible to determine the root path of an deep nested structure when the value is also a deep nested structure.

`Result` contains the wrapped data extracted from a specific annotation key/value pair in `Object`, either `bindingType("env")` and `bindingType("volumemount")` or any other types it might support in the future in `Type` and the extracted value's root path in `Path`